examples/stein_dmm.py

```diff
@@ -0,0 +1,339 @@
+
+        with numpyro.handlers.scale(scale=annealing_factor):
+            # Actually score 'z'
+            numpyro.sample(
+                "z_aux",
+                dist.Normal(z_loc, z_scale)
+                .mask(jnp.expand_dims(masks, axis=-1))
+                .to_event(2),
+                obs=z,
+            )
+
+        emission_probs = emitter(z, params=emitter_params)
+        if predict:
+            tunes = None
+        else:
+            tunes = seqs_batch
+        numpyro.sample(
+            "tunes",
+            dist.Bernoulli(logits=emission_probs)
+            .mask(jnp.expand_dims(masks, axis=-1))
+            .to_event(2),
+            obs=tunes,
+        )
+
+
+def guide(
+    seqs,
+    seqs_rev,
+    lengths,
+    *,
+    subsample_size=77,
+    latent_dim=32,
+    emission_dim=100,
+    transition_dim=200,
+    data_dim=88,
+    gru_dim=150,
+    annealing_factor=1.0,
+    predict=False,
+):
+    max_seq_length = seqs.shape[1]
+    seqs_rev = jnp.transpose(seqs_rev, axes=(1, 0, 2))
+
+    combiner_params = {
+        "mean": numpyro.param("combiner_mean", _normal_init(gru_dim, latent_dim)),
+        "std": numpyro.param("combiner_std", _normal_init(gru_dim, latent_dim)),
+    }
+
+    gru_params = {
+        "update_in": numpyro.param("update_in", _normal_init(data_dim, gru_dim)),
+        "update_weight": numpyro.param("update_weight", _normal_init(gru_dim, gru_dim)),
+        "reset_in": numpyro.param("reset_in", _normal_init(data_dim, gru_dim)),
+        "reset_weight": numpyro.param("reset_weight", _normal_init(gru_dim, gru_dim)),
+        "out_in": numpyro.param("out_in", _normal_init(data_dim, gru_dim)),
+        "out_weight": numpyro.param("out_weight", _normal_init(gru_dim, gru_dim)),
+    }
+
+    with numpyro.plate(
+        "data", seqs.shape[0], subsample_size=subsample_size, dim=-1
+    ) as idx:
+        if subsample_size == seqs.shape[0]:
+            seqs_rev_batch = seqs_rev
+            lengths_batch = lengths
+        else:
+            seqs_rev_batch = seqs_rev[:, idx, :]
+            lengths_batch = lengths[idx]
+
+        masks = jnp.repeat(
+            jnp.expand_dims(jnp.arange(max_seq_length), axis=0), subsample_size, axis=0
+        ) < jnp.expand_dims(lengths_batch, axis=-1)
+
+        h0 = numpyro.param(
+            "h0",
+            lambda rng_key: dist.Normal(0.0, 1).sample(rng_key, (1, gru_dim)),
+        )
+        _, hs = gru(seqs_rev_batch, lengths_batch, h0, gru_params)
+        hs = _reverse_padded(jnp.transpose(hs, axes=(1, 0, 2)), lengths_batch)
+        with numpyro.handlers.scale(scale=annealing_factor):
+            numpyro.sample(
+                "z",
+                dist.Normal(*combiner(hs, combiner_params))
+                .mask(jnp.expand_dims(masks, axis=-1))
+                .to_event(2),
+            )
+
+
+def vis_tune(i, tunes, lengths, name="stein_dmm.pdf"):
+    try:
+        from music21.chord import Chord
```

Comment:
Would it be preferable to use imshow (and not introduce a dependency)?

Reply:
I think this `try except` pattern for optional dependency is fine.